Ensure updates with stable_comment are pushed to stable automatically

mattiaverga opened the pull request:
The `approve_testing` task was set to ignore updates which already had
stable_comment.
This was not correct and prevented some updates from being pushed to
stable when they reached the stable_days threshold.

As an example, consider an update for which the associated release has a
mandatory_days_in testing set to 7 and the update is created with a
stable_days threshold of 10.
After 7 days in testing, the update gets the stable_comment, since the
maintainer can now push it to stable.
The autopush is performed after 3 more days, but this didn't happen
because of the approve_testing task ignoring all updates with
stable_comment.

Fix #4042 

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

mattiaverga committed:
Ensure updates with stable_comment are pushed to stable automatically

The `approve_testing` task was set to ignore updates which already had
stable_comment.
This was not correct and prevented some updates from being pushed to
stable when they reached the stable_days threshold.

As an example, consider an update for which the associated release has a
mandatory_days_in testing set to 7 and the update is created with a
stable_days threshold of 10.
After 7 days in testing, the update gets the stable_comment, since the
maintainer can now push it to stable.
The autopush is performed after 3 more days, but this didn't happen
because of the approve_testing task ignoring all updates with
stable_comment.

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

pypingou commented:
I would recommend describing more in the commit message how we ended up in this situation, but I leave this up to you and if you don't want I'll approve anyway

mattiaverga committed:
Clarify when an update can be pushed to stable

There's a lot of confusion between the required thresholds to push an
update to stable and the thresholds set for autopush. The documentation
lacks an appropriate description.

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

mattiaverga commented:
I tried to better describe the case, see if it fits now.
I've also added a section in the docs about the differences between the mandatory thresholds and the user defined thresholds for autopush, I saw a lot of confusion in the past about that...

pypingou reviewed (approved):
Thanks for the commit message, I understand the issue addressed here much better :)

:+1: on the doc update as well

ryanlerch committed:
Merge branch 'develop' into autopush
